tests/tests_pytorch/trainer/logging_/test_train_loop_logging.py

```diff
@@ -621,11 +622,26 @@ def training_step(self, *args):
 
     trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=1)
     model = TestModel()
-    match = escape(f"self.log(foo, {value})` was called")
+    match = escape(f"self.log(foo, {value})` was called, but the tensor must have a single element")
     with pytest.raises(ValueError, match=match):
         trainer.fit(model)
 
 
+def test_log_tensor_and_clone_no_torch_warning(tmpdir):
+    """Regression test for issue https://github.com/Lightning-AI/lightning/issues/14594."""
+
+    class TestModel(BoringModel):
+        def training_step(self, *args):
+            self.log("foo", torch.tensor(1))
+            return super().training_step(*args)
+
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=1)
+    model = TestModel()
+    match = escape("To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach()")
```

Comment:
I suggest making this text smaller so that a change in PyTorch to the text does not make this test useless.

```suggestion
    match = escape("recommended.*.clone().detach()")
```

(I didn't test if escape works with regex `.*`), might need tweaking

Reply:
@awaelchli - If you agree to this change, can you please commit? Or if it's resolved offline, please feel free to resolve this conversation. Let's try to get this PR to today's release 🚀